controllers/octavia_controller.go

```diff
@@ -248,37 +250,31 @@ func (r *OctaviaReconciler) reconcileInit(
 	//
 	// create service DB instance
 	//
-	octaviaDb := mariadbv1.NewDatabase(
-		octavia.DatabaseName,
-		instance.Spec.DatabaseUser,
-		instance.Spec.Secret,
-		map[string]string{
-			"dbName": instance.Spec.DatabaseInstance,
-		},
+	octaviaDb := mariadbv1.NewDatabaseForAccount(
+		instance.Spec.DatabaseInstance, // mariadb/galera service to target
+		instance.Name,                  // name used in CREATE DATABASE in mariadb
+		instance.Name,                  // CR name for MariaDBDatabase
+		instance.Spec.DatabaseAccount,  // CR name for MariaDBAccount
+		instance.Namespace,             // namespace
 	)
 	// Using WithNamespace to explicitly set a name that is not "octavia"
 	// If name is not set, it's set automatically to the name of the service ("octavia"),
 	// and it triggers a conflict in the mariadbdatabase functions when two DBs
 	// share the same "name"
-	persistentDb := mariadbv1.NewDatabaseWithNamespace(
-		octavia.PersistenceDatabaseName,
-		instance.Spec.DatabaseUser,
-		instance.Spec.Secret,
-		map[string]string{
-			"dbName": instance.Spec.DatabaseInstance,
-		},
-		"octavia-persistence",
-		instance.Namespace,
+	persistentDb := mariadbv1.NewDatabaseForAccount(
+		instance.Spec.DatabaseInstance,  // mariadb/galera service to target
+		octavia.PersistenceDatabaseName, // name used in CREATE DATABASE in mariadb
+		"octavia-persistence",           // CR name for MariaDBDatabase
+		instance.Spec.DatabaseAccount,   // CR name for MariaDBAccount
```

Comment:
OK, this is the problem, or at least one of them, a MariaDBAccount is only for one MariaDBDatabase. If octavia has two databases, we need two MariaDBAccounts